src/types/errors.d.ts

```diff
@@ -4,6 +4,7 @@ export type FaunaHttpErrorResponseContent = {
   errors: {
     code: string,
     description: string
+    cause: any;
```

Comment:
The cause field is an array of additional errors, recursively, so I think its type can be specified like this

```suggestion
    cause?: FaunaHttpErrorResponseContent[];
```

Here's an example raw response from a query where a UDF calls a UDF which then has an error:

```json
{
  "errors": [
    {
      "position": [  ],
      "code": "call error",
      "description": "Calling the function resulted in an error.",
      "cause": [
        {
          "position": [ "expr" ],
          "code": "call error",
          "description": "Calling the function resulted in an error.",
          "cause": [
            {
              "position": [ "expr", "concat", 0 ],
              "code": "invalid argument",
              "description": "String expected, Integer provided."
            },
            {
              "position": [ "expr", "concat", 1 ],
              "code": "invalid argument",
              "description": "String expected, Integer provided."
            }
          ]
        }
      ]
    }
  ]
}
```

Do you have any other examples?